rebasehelper/build_log_hooks/files_build_log_hook.py

```diff
@@ -0,0 +1,198 @@
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Authors: Petr Hracek <phracek@redhat.com>
+#          Tomas Hozza <thozza@redhat.com>
+
+import difflib
+import fnmatch
+import os
+import re
+
+import six
+
+from rebasehelper.build_log_hook import BaseBuildLogHook
+from rebasehelper.helpers.macro_helper import MacroHelper
+from rebasehelper.logger import logger
+from rebasehelper.constants import FILES_DIRECTIVES
+
+
+class FilesBuildLogHook(BaseBuildLogHook):
+    NAME = 'files'
+    CATEGORIES = None
+
+    @classmethod
+    def get_name(cls):
+        return cls.NAME
+
+    @classmethod
+    def get_categories(cls):
+        return cls.CATEGORIES
+
+    @classmethod
+    def _parse_build_log(cls, log_path, nvr):
+        """Parses a build log.
+
+        Args:
+            log_path (str): Path to the RPM build log.
+
+        Returns:
+            tuple: The first element is the type of error that was found
+                in the log (missing or deleted). The second element is
+                a list of problematic files.
+
+        """
+        try:
+            with open(log_path, 'r') as build_log:
+                lines = build_log.read().splitlines()
+        except IOError:
+            logger.info('There was an error opening %s', log_path)
+            return None, None
+
+        error_type = None
+        # Use set to avoid duplicate files.
+        files = set()
+
+        # Two types of error can occur. The file is either in the SPEC file but missing in sources
+        # or the other way around. Example lines of the first case:
+        # File not found: /root/rpmbuild/BUILDROOT/pello-0.1.1-1.fc27.x86_64/should/not/be/here
+        # File not found: /root/rpmbuild/BUILDROOT/pello-0.1.1-1.fc27_x86_64/should/not/be/here2
+        # Example lines of the second case:
+        # Installed (but unpackaged) file(s) found:
+        #   /usr/bin/pello
+        error_re = re.compile(r'^(BUILDSTDERR:)?\s*((?P<missing>File not found:\s*)|'
+                              r'(?P<unpackaged>Installed \(but unpackaged\) file\(s\) found:))?'
+                              r'(/.*/{}[\.\w]+)?(?P<path>/.*)?$'.format(re.escape(nvr)))
+        for line in lines:
+            match = re.match(error_re, line)
+            if match:
+                if match.group('missing'):
+                    error_type = 'deleted'
+                    files.add(match.group('path'))
+                elif match.group('unpackaged'):
+                    error_type = 'missing'
+                elif error_type == 'missing' and match.group('path'):
+                    files.add(match.group('path'))
+                elif error_type and not match.group('path'):
+                    break
+
+        return error_type, list(files)
+
+    @classmethod
+    def _get_best_matching_files_section(cls, rebase_spec_file, file):
+        """Finds a section with a file that has the closest match with
+        the given file.
```

Comment:
This formulation is rather confusing. How about `Finds a %files section containing a file that has the closest match with the specified file`?